iac-extensions/arm/src/test/java/org/sonar/iac/arm/tree/impl/json/ArrayExpressionImplTest.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.arm.tree.impl.json;
+
+import org.junit.jupiter.api.Test;
+import org.sonar.iac.arm.parser.ArmParser;
+import org.sonar.iac.arm.tree.api.ArrayExpression;
+import org.sonar.iac.arm.tree.api.Property;
+
+import static org.sonar.iac.arm.ArmAssertions.assertThat;
+import static org.sonar.iac.arm.tree.impl.json.PropertyTestUtils.LINE_OFFSET;
+import static org.sonar.iac.arm.tree.impl.json.PropertyTestUtils.parseProperty;
+
+class ArrayExpressionImplTest {
+  private final ArmParser parser = new ArmParser();
+
+  @Test
+  void shouldParseArrayExpression() {
+    Property arrayProperty = parseProperty(parser, "\"array_prop\": [\"val\"]");
+    assertThat(arrayProperty.value())
+      .asArrayExpression()
+      .containsValuesExactly("val")
+      .hasRange(LINE_OFFSET + 1, 14, LINE_OFFSET + 1, 21);
+  }
+
+  @Test
+  void shouldParseEmptyArrayExpression() {
+    Property arrayProperty = parseProperty(parser, "\"array_prop\": []");
+    assertThat(arrayProperty.value()).asArrayExpression().containsValuesExactly();
+  }
+
+  @Test
+  void shouldParseArrayExpressionFilledWithDifferentTypesOfValues() {
+    Property arrayProperty = parseProperty(parser, "\"array_prop\": [\"val\", 5, {\"key\":\"val\"}, [\"val\"]]");
+    assertThat(arrayProperty.value()).asArrayExpression();
```

Comment:
IMO this line is not needed, the missing assignment looks strange (I know there is a check also under the hood).
If the value will be not an `ArrayExpression`, the exception from the next line will inform us about it.